app/services/finance/ecf/assurance_report/query.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module Finance
+  module ECF
+    class AssuranceReport
+      class Query
+        def initialize(statement)
```

Comment:
Out of curiosity did you run those queries in prod, just curious about it's performance?

Reply:
yes I do run those every output fee statement. Pretty fast.